zcash_client_sqlite/src/lib.rs

```diff
@@ -525,17 +525,21 @@ impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
 /// A wrapper for the SQLite connection to the block cache database.
 pub struct BlockDb(Connection);
 
+unsafe impl Send for BlockDb {}
+unsafe impl Sync for BlockDb {}
+
 impl BlockDb {
     /// Opens a connection to the wallet database stored at the specified path.
     pub fn for_path<P: AsRef<Path>>(path: P) -> Result<Self, rusqlite::Error> {
         Connection::open(path).map(BlockDb)
     }
 }
 
+#[async_trait::async_trait(?Send)]
```

Comment:
Did you try to sue just #[async_trait] annotation and add `+ Send` bound to `F`? Or was that the cause of the lifetime problem?

Reply:
yes I was going to do that but it's produces some weird lifetime errors, and so this is me trying to ignore the `Send` requirements by making it optional